catalyst/dl/callbacks/base.py

```diff
@@ -135,6 +135,65 @@ def on_stage_end(self, state):
         print(top_best_metrics_str)
 
 
+class IterationCheckpointCallback(Callback):
+    """
+    Iteration checkpoint callback to save your model/criterion/optimizer/metrics
+    """
+
+    def __init__(
+        self, n_iters : int = 100, stage_restart : bool = True
```

Comment:
please, use `num_iters` instead, as we do with `num_workers`/`num_epochs`/etc

Reply:
and we need something like `save_n_best` to reduce disk usage (if you have large model and a lot of data, you can occupy all the HDD space in this way)

Reply:
> and we need something like save_n_best to reduce disk usage (if you have large model and a lot of data, you can occupy all the HDD space in this way)

I think `save_n_last` is better. If we use `save_n_best`, every time we save the iteration checkpoint, we should evaluate the validation again. 
What do you think?

The main purpose that I add this callback is, 
Suppose that I have 4M images to train, one epoch takes several days to train. The checkpoint is only saved at the end of the epoch. This is quite risky for me if our server is turn-off suddenly then I lose everything. Therefore, Iteration callback can save me this problem.

Reply:
Deal, let's use `save_n_last` approach.